framework/src/reporters/ConstantReporter.C

```diff
@@ -18,82 +18,45 @@ ConstantReporter::validParams()
-  std::vector<std::string> string_val = getParam<std::vector<std::string>>("string_values");
+  params.addParam<std::vector<ReporterValueName>>("string_names", "Names for each string value.");
+  params.addParam<std::vector<std::string>>("string_values", "Values for strings.");
 
-  unsigned int ii = 0;
-  unsigned int ir = 0;
-  unsigned int iv = 0;
-  unsigned int is = 0;
-  for (unsigned int i = 0; i < names.size(); ++i)
-  {
-    if (types[i] == "integer")
-    {
-      if (ii >= int_val.size())
-        paramError("integer_values", "Not enough values provided.");
-      _int.push_back(&declareValueByName<int>(names[i], int_val[ii++]));
-    }
+  // Vector data
+  params.addParam<std::vector<ReporterValueName>>("integer_vector_names",
+                                                  "Names for each vector of integers value.");
+  params.addParam<std::vector<std::vector<int>>>("integer_vector_values",
+                                                 "Values for vectors of integers.");
 
-    else if (types[i] == "real")
-    {
-      if (ir >= real_val.size())
-        paramError("real_values", "Not enough values provided.");
-      _real.push_back(&declareValueByName<Real>(names[i], real_val[ir++]));
-    }
+  params.addParam<std::vector<ReporterValueName>>("real_vector_names",
+                                                  "Names for each vector of reals value.");
+  params.addParam<std::vector<std::vector<Real>>>("real_vector_values",
+                                                  "Values for vectors of reals.");
 
-    else if (types[i] == "vector")
-    {
-      if (iv >= vector_val.size())
-        paramError("vector_values", "Not enough values provided.");
-      _vector.push_back(&declareValueByName<std::vector<Real>>(names[i], vector_val[iv++]));
-    }
+  params.addParam<std::vector<ReporterValueName>>("string_vector_names",
+                                                  "Names for each vector of strings value.");
+  params.addParam<std::vector<std::vector<std::string>>>("string_vector_values",
+                                                         "Values for vectors of strings.");
 
-    else if (types[i] == "string")
-    {
-      if (is >= string_val.size())
-        paramError("string_values", "Not enough values provided.");
-      _string.push_back(&declareValueByName<std::string>(names[i], string_val[is++]));
-    }
-  }
+  return params;
+}
 
-  if (ii < int_val.size())
-    paramError("integer_values", "Unused values.");
-  if (ir < real_val.size())
-    paramError("real_values", "Unused values.");
-  if (iv < vector_val.size())
-    paramError("vector_values", "Unused values.");
-  if (is < string_val.size())
-    paramError("string_values", "Unused values.");
+ConstantReporter::ConstantReporter(const InputParameters & parameters)
+  : GeneralReporter(parameters),
+    _int(declareConstantReporterValues<int>("integer_names", "integer_values")),
```

Comment:
Based on my comment above this would then become:
```
_int(declareConstantReporterValues<int>("integer")),
```

Reply:
Just an idea, if you like it go ahead and change it. If you don't we can leave it as is.